tests/Bedrock.Framework.Benchmarks/HttpHeaderReaderBenchmarks.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Bedrock.Framework.Protocols.Http.Http1;
+using BenchmarkDotNet.Attributes;
+using System;
+using System.Buffers;
+using System.Text;
+
+namespace Bedrock.Framework.Benchmarks
+{
+    public class HttpHeaderReaderBenchmarks
+    {
+        [Params(10, 1000)]
+        public int HeaderValueLength { get; set; }
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            var data = "/search?rlz=1C1GCEA_enIL874IL874&sxsrf=ACYBGNQAjkw0VCQs1ElFY43_D_Wq9DMwQA%3A1580624707397&ei=Q2s2XtjxF8_RwAK35ocQ&q=request+header+example&oq=request+header+example&gs_l=psy-ab.3..0i7i30l10.9636.9636..9981...0.2..0.245.245.2-1......0....1..gws-wiz.......0i71.JfZoTvof620&ved=0ahUKEwiYn9TxnbLnAhXPKFAKHTfzAQIQ4dUDCAs&uact=5/search?rlz=1C1GCEA_enIL874IL874&sxsrf=ACYBGNQAjkw0VCQs1ElFY43_D_Wq9DMwQA%3A1580624707397&ei=Q2s2XtjxF8_RwAK35ocQ&q=request+header+example&oq=request+header+example&gs_l=psy-ab.3..0i7i30l10.9636.9636..9981...0.2..0.245.245.2-1......0....1..gws-wiz.......0i71.JfZoTvof620&ved=0ahUKEwiYn9TxnbLnAhXPKFAKHTfzAQIQ4dUDCAs&uact=5/search?rlz=1C1GCEA_enIL874IL874&sxsrf=ACYBGNQAjkw0VCQs1ElFY43_D_Wq9DMwQA%3A1580624707397&ei=Q2s2XtjxF8_RwAK35ocQ&q=request+header+example&oq=request+header+example&gs_l=psy-ab.3..0i7i30l10.9636.9636..9981...0.2..0.245.245.2-1......0....1..gws-wiz.......0i71.JfZoTvof620&ved=0ahUKEwiYn9TxnbLnAhXPKFAKHTfzAQIQ4dUDCAs&uact=5/search?rlz=1C1GCEA_enIL874IL874&sxsrf=ACYBGNQAjkw0VCQs1ElFY43_D_Wq9DMwQA%3A1580624707397&ei=Q2s2XtjxF8_RwAK35ocQ&q=request+header+example&oq=request+header+example&gs_l=psy-ab.3..0i7i30l10.9636.9636..9981...0.2..0.245.245.2-1......0....1..gws-wiz.......0i71.JfZoTvof620&ved=0ahUKEwiYn9TxnbLnAhXPKFAKHTfzAQIQ4dUDCAs&uact=5";
```

Comment:
Where did you get this header?

Reply:
It was the longest header from a Google search I did, repeated 3 times over...

Reply:
OK, it's fine for now I guess just not your typical header use case.